### PR TITLE
security: Update rust crate versions

### PR DESCRIPTION
**Pull request by jodh-intel**
security: Update rust crate versions

Update the rust dependencies that have upstream security fixes. Issues
fixed by this change:

- [`RUSTSEC-2020-0002`](https://rustsec.org/advisories/RUSTSEC-2020-0002) (`prost` crate)
- [`RUSTSEC-2020-0036`](https://rustsec.org/advisories/RUSTSEC-2020-0036) (`failure` crate)
- [`RUSTSEC-2021-0073`](https://rustsec.org/advisories/RUSTSEC-2021-0073) (`prost-types` crate)
- [`RUSTSEC-2021-0119`](https://rustsec.org/advisories/RUSTSEC-2021-0119) (`nix` crate)

This change also includes:

- Minor code changes for the new version of `prometheus` for the agent.

- A *downgrade* of the version of the `futures` crate to the (new)
latest version (`0.3.17`) since version `0.3.18` was removed [1].

Fixes: #3296.

[1] - See https://crates.io/crates/futures/versions

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: src/agent/rustjail/Cargo.toml [hunk 1]
```diff
@@ -23,11 +23,11 @@ scan_fmt = "0.2.6"
 regex = "1.5.4"
 path-absolutize = "1.2.0"
 anyhow = "1.0.32"
-cgroups = { package = "cgroups-rs", version = "0.2.5" }
+cgroups = { package = "cgroups-rs", version = "0.2.8" }
 rlimit = "0.5.3"
 
 tokio = { version = "1.2.0", features = ["sync", "io-util", "process", "time", "macros"] }
-futures = "0.3.18"
+futures = "0.3.17"
```
Comment (liubin) [inline src/agent/rustjail/Cargo.toml:30]: From 0.3.18 to 0.3.17?

Comment (jodh-intel) [reply src/agent/rustjail/Cargo.toml:30]: Unfortunately yes - that version was removed - see https://crates.io/crates/futures/versions.

**Review by Jakob-Naucke — APPROVED** [review]
Changes look good. AFAICT, `protoc` is now required for all architectures (used to be ppc64le/s390x only).

**Review by bergwolf — APPROVED** [review]
Thanks a lot @jodh-intel !

**Comment by jodh-intel** [comment]
Here's the output of `cargo audit -c never -D warnings` on `main`:

```
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 378 security advisories (from /home/johunt/.cargo/advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (192 crate dependencies)
Crate:         chrono
Version:       0.4.19
Title:         Potential segfault in `localtime_r` invocations
Date:          2020-11-10
ID:            RUSTSEC-2020-0159
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0159
Solution:      No safe upgrade is available!
Dependency tree: 
chrono 0.4.19
├── tracing-subscriber 0.2.25
│   ├── tracing-opentelemetry 0.13.0
│   │   └── kata-agent 0.1.0
│   └── kata-agent 0.1.0
├── slog-json 2.4.0
│   └── logging 0.1.0
│       └── kata-agent 0.1.0
└── procfs 0.7.9
    ├── prometheus 0.9.0
    │   └── kata-agent 0.1.0
    └── kata-agent 0.1.0

Crate:         nix
Version:       0.16.1
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.16.1

Crate:         nix
Version:       0.17.0
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.17.0

Crate:         nix
Version:       0.19.1
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.19.1

Crate:         nix
Version:       0.20.0
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.20.0

Crate:         nix
Version:       0.22.0
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.22.0

Crate:         prost
Version:       0.5.0
Title:         Parsing a specially crafted message can result in a stack overflow
Date:          2020-01-16
ID:            RUSTSEC-2020-0002
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0002
Solution:      Upgrade to >=0.6.1
Dependency tree: 
prost 0.5.0
├── ttrpc-compiler 0.4.0
│   └── ttrpc-codegen 0.2.0
│       └── protocols 0.1.0
│           ├── rustjail 0.1.0
│           │   └── kata-agent 0.1.0
│           └── kata-agent 0.1.0
├── prost-types 0.5.0
│   ├── ttrpc-compiler 0.4.0
│   └── prost-build 0.5.0
│       └── ttrpc-compiler 0.4.0
└── prost-build 0.5.0

Crate:         prost-types
Version:       0.5.0
Title:         Conversion from `prost_types::Timestamp` to `SystemTime` can cause an overflow and panic
Date:          2021-07-08
ID:            RUSTSEC-2021-0073
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0073
Solution:      Upgrade to >=0.8.0
Dependency tree: 
prost-types 0.5.0
├── ttrpc-compiler 0.4.0
│   └── ttrpc-codegen 0.2.0
│       └── protocols 0.1.0
│           ├── rustjail 0.1.0
│           │   └── kata-agent 0.1.0
│           └── kata-agent 0.1.0
└── prost-build 0.5.0
    └── ttrpc-compiler 0.4.0

Crate:         time
Version:       0.1.43
Title:         Potential segfault in the time crate
Date:          2020-11-18
ID:            RUSTSEC-2020-0071
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0071
Solution:      Upgrade to >=0.2.23
Dependency tree: 
time 0.1.43
└── chrono 0.4.19
    ├── tracing-subscriber 0.2.25
    │   ├── tracing-opentelemetry 0.13.0
    │   │   └── kata-agent 0.1.0
    │   └── kata-agent 0.1.0
    ├── slog-json 2.4.0
    │   └── logging 0.1.0
    │       └── kata-agent 0.1.0
    └── procfs 0.7.9
        ├── prometheus 0.9.0
        │   └── kata-agent 0.1.0
        └── kata-agent 0.1.0

Crate:         failure
Version:       0.1.8
Warning:       unmaintained
Title:         failure is officially deprecated/unmaintained
Date:          2020-05-02
ID:            RUSTSEC-2020-0036
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0036
Dependency tree: 
failure 0.1.8
├── which 2.0.1
│   └── prost-build 0.5.0
│       └── ttrpc-compiler 0.4.0
│           └── ttrpc-codegen 0.2.0
│               └── protocols 0.1.0
│                   ├── rustjail 0.1.0
│                   │   └── kata-agent 0.1.0
│                   └── kata-agent 0.1.0
└── prost-derive 0.5.0
    └── prost 0.5.0
        ├── ttrpc-compiler 0.4.0
        ├── prost-types 0.5.0
        │   ├── ttrpc-compiler 0.4.0
        │   └── prost-build 0.5.0
        └── prost-build 0.5.0

Crate:         failure
Version:       0.1.8
Warning:       unsound
Title:         Type confusion if __private_get_type_id__ is overridden
Date:          2019-11-13
ID:            RUSTSEC-2019-0036
URL:           https://rustsec.org/advisories/RUSTSEC-2019-0036

Crate:         futures
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures 0.3.18
├── ttrpc 0.5.1
│   ├── protocols 0.1.0
│   │   ├── rustjail 0.1.0
│   │   │   └── kata-agent 0.1.0
│   │   └── kata-agent 0.1.0
│   └── kata-agent 0.1.0
├── tokio-vsock 0.3.1
│   ├── vsock-exporter 0.1.0
│   │   └── kata-agent 0.1.0
│   ├── ttrpc 0.5.1
│   └── kata-agent 0.1.0
├── rustjail 0.1.0
├── rtnetlink 0.8.1
│   └── kata-agent 0.1.0
├── opentelemetry 0.14.0
│   ├── vsock-exporter 0.1.0
│   ├── tracing-opentelemetry 0.13.0
│   │   └── kata-agent 0.1.0
│   └── kata-agent 0.1.0
├── netlink-sys 0.7.0
│   ├── netlink-proto 0.7.0
│   │   └── rtnetlink 0.8.1
│   └── kata-agent 0.1.0
├── netlink-proto 0.7.0
└── kata-agent 0.1.0

Crate:         futures-channel
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-channel 0.3.18
├── futures-util 0.3.18
│   ├── futures-executor 0.3.18
│   │   └── futures 0.3.18
│   │       ├── ttrpc 0.5.1
│   │       │   ├── protocols 0.1.0
│   │       │   │   ├── rustjail 0.1.0
│   │       │   │   │   └── kata-agent 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── tokio-vsock 0.3.1
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   ├── ttrpc 0.5.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── rustjail 0.1.0
│   │       ├── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── opentelemetry 0.14.0
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   ├── tracing-opentelemetry 0.13.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-sys 0.7.0
│   │       │   ├── netlink-proto 0.7.0
│   │       │   │   └── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-proto 0.7.0
│   │       └── kata-agent 0.1.0
│   └── futures 0.3.18
└── futures 0.3.18

Crate:         futures-core
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-core 0.3.18
├── tokio-util 0.6.9
│   └── netlink-proto 0.7.0
│       └── rtnetlink 0.8.1
│           └── kata-agent 0.1.0
├── tokio-stream 0.1.8
│   └── opentelemetry 0.14.0
│       ├── vsock-exporter 0.1.0
│       │   └── kata-agent 0.1.0
│       ├── tracing-opentelemetry 0.13.0
│       │   └── kata-agent 0.1.0
│       └── kata-agent 0.1.0
├── inotify 0.9.6
│   └── rustjail 0.1.0
│       └── kata-agent 0.1.0
├── futures-util 0.3.18
│   ├── futures-executor 0.3.18
│   │   └── futures 0.3.18
│   │       ├── ttrpc 0.5.1
│   │       │   ├── protocols 0.1.0
│   │       │   │   ├── rustjail 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── tokio-vsock 0.3.1
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   ├── ttrpc 0.5.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── rustjail 0.1.0
│   │       ├── rtnetlink 0.8.1
│   │       ├── opentelemetry 0.14.0
│   │       ├── netlink-sys 0.7.0
│   │       │   ├── netlink-proto 0.7.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-proto 0.7.0
│   │       └── kata-agent 0.1.0
│   └── futures 0.3.18
├── futures-executor 0.3.18
├── futures-channel 0.3.18
│   ├── futures-util 0.3.18
│   └── futures 0.3.18
└── futures 0.3.18

Crate:         futures-executor
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-executor 0.3.18
└── futures 0.3.18
    ├── ttrpc 0.5.1
    │   ├── protocols 0.1.0
    │   │   ├── rustjail 0.1.0
    │   │   │   └── kata-agent 0.1.0
    │   │   └── kata-agent 0.1.0
    │   └── kata-agent 0.1.0
    ├── tokio-vsock 0.3.1
    │   ├── vsock-exporter 0.1.0
    │   │   └── kata-agent 0.1.0
    │   ├── ttrpc 0.5.1
    │   └── kata-agent 0.1.0
    ├── rustjail 0.1.0
    ├── rtnetlink 0.8.1
    │   └── kata-agent 0.1.0
    ├── opentelemetry 0.14.0
    │   ├── vsock-exporter 0.1.0
    │   ├── tracing-opentelemetry 0.13.0
    │   │   └── kata-agent 0.1.0
    │   └── kata-agent 0.1.0
    ├── netlink-sys 0.7.0
    │   ├── netlink-proto 0.7.0
    │   │   └── rtnetlink 0.8.1
    │   └── kata-agent 0.1.0
    ├── netlink-proto 0.7.0
    └── kata-agent 0.1.0

Crate:         futures-io
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-io 0.3.18
├── futures-util 0.3.18
│   ├── futures-executor 0.3.18
│   │   └── futures 0.3.18
│   │       ├── ttrpc 0.5.1
│   │       │   ├── protocols 0.1.0
│   │       │   │   ├── rustjail 0.1.0
│   │       │   │   │   └── kata-agent 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── tokio-vsock 0.3.1
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   ├── ttrpc 0.5.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── rustjail 0.1.0
│   │       ├── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── opentelemetry 0.14.0
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   ├── tracing-opentelemetry 0.13.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-sys 0.7.0
│   │       │   ├── netlink-proto 0.7.0
│   │       │   │   └── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-proto 0.7.0
│   │       └── kata-agent 0.1.0
│   └── futures 0.3.18
└── futures 0.3.18

Crate:         futures-sink
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-sink 0.3.18
├── tokio-util 0.6.9
│   └── netlink-proto 0.7.0
│       └── rtnetlink 0.8.1
│           └── kata-agent 0.1.0
├── futures-util 0.3.18
│   ├── futures-executor 0.3.18
│   │   └── futures 0.3.18
│   │       ├── ttrpc 0.5.1
│   │       │   ├── protocols 0.1.0
│   │       │   │   ├── rustjail 0.1.0
│   │       │   │   │   └── kata-agent 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── tokio-vsock 0.3.1
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   ├── ttrpc 0.5.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── rustjail 0.1.0
│   │       ├── rtnetlink 0.8.1
│   │       ├── opentelemetry 0.14.0
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   ├── tracing-opentelemetry 0.13.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-sys 0.7.0
│   │       │   ├── netlink-proto 0.7.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-proto 0.7.0
│   │       └── kata-agent 0.1.0
│   └── futures 0.3.18
├── futures-channel 0.3.18
│   ├── futures-util 0.3.18
│   └── futures 0.3.18
└── futures 0.3.18

Crate:         futures-task
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-task 0.3.18
├── futures-util 0.3.18
│   ├── futures-executor 0.3.18
│   │   └── futures 0.3.18
│   │       ├── ttrpc 0.5.1
│   │       │   ├── protocols 0.1.0
│   │       │   │   ├── rustjail 0.1.0
│   │       │   │   │   └── kata-agent 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── tokio-vsock 0.3.1
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   ├── ttrpc 0.5.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── rustjail 0.1.0
│   │       ├── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── opentelemetry 0.14.0
│   │       │   ├── vsock-exporter 0.1.0
│   │       │   ├── tracing-opentelemetry 0.13.0
│   │       │   │   └── kata-agent 0.1.0
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-sys 0.7.0
│   │       │   ├── netlink-proto 0.7.0
│   │       │   │   └── rtnetlink 0.8.1
│   │       │   └── kata-agent 0.1.0
│   │       ├── netlink-proto 0.7.0
│   │       └── kata-agent 0.1.0
│   └── futures 0.3.18
├── futures-executor 0.3.18
└── futures 0.3.18

Crate:         futures-util
Version:       0.3.18
Warning:       yanked
Dependency tree: 
futures-util 0.3.18
├── futures-executor 0.3.18
│   └── futures 0.3.18
│       ├── ttrpc 0.5.1
│       │   ├── protocols 0.1.0
│       │   │   ├── rustjail 0.1.0
│       │   │   │   └── kata-agent 0.1.0
│       │   │   └── kata-agent 0.1.0
│       │   └── kata-agent 0.1.0
│       ├── tokio-vsock 0.3.1
│       │   ├── vsock-exporter 0.1.0
│       │   │   └── kata-agent 0.1.0
│       │   ├── ttrpc 0.5.1
│       │   └── kata-agent 0.1.0
│       ├── rustjail 0.1.0
│       ├── rtnetlink 0.8.1
│       │   └── kata-agent 0.1.0
│       ├── opentelemetry 0.14.0
│       │   ├── vsock-exporter 0.1.0
│       │   ├── tracing-opentelemetry 0.13.0
│       │   │   └── kata-agent 0.1.0
│       │   └── kata-agent 0.1.0
│       ├── netlink-sys 0.7.0
│       │   ├── netlink-proto 0.7.0
│       │   │   └── rtnetlink 0.8.1
│       │   └── kata-agent 0.1.0
│       ├── netlink-proto 0.7.0
│       └── kata-agent 0.1.0
└── futures 0.3.18

error: 9 vulnerabilities found!
error: 10 denied warnings found!
```

And here's the output of the same command with this PR applied:

```
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 378 security advisories (from /home/johunt/.cargo/advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (181 crate dependencies)
Crate:         chrono
Version:       0.4.19
Title:         Potential segfault in `localtime_r` invocations
Date:          2020-11-10
ID:            RUSTSEC-2020-0159
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0159
Solution:      No safe upgrade is available!
Dependency tree: 
chrono 0.4.19
├── tracing-subscriber 0.2.25
│   ├── tracing-opentelemetry 0.13.0
│   │   └── kata-agent 0.1.0
│   └── kata-agent 0.1.0
├── slog-json 2.4.0
│   └── logging 0.1.0
│       └── kata-agent 0.1.0
└── procfs 0.12.0

Crate:         nix
Version:       0.17.0
Title:         Out-of-bounds write in nix::unistd::getgrouplist
Date:          2021-09-27
ID:            RUSTSEC-2021-0119
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0119
Solution:      Upgrade to ^0.20.2 OR ^0.21.2 OR ^0.22.2 OR >=0.23.0
Dependency tree: 
nix 0.17.0

Crate:         time
Version:       0.1.43
Title:         Potential segfault in the time crate
Date:          2020-11-18
ID:            RUSTSEC-2020-0071
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0071
Solution:      Upgrade to >=0.2.23
Dependency tree: 
time 0.1.43
└── chrono 0.4.19
    ├── tracing-subscriber 0.2.25
    │   ├── tracing-opentelemetry 0.13.0
    │   │   └── kata-agent 0.1.0
    │   └── kata-agent 0.1.0
    ├── slog-json 2.4.0
    │   └── logging 0.1.0
    │       └── kata-agent 0.1.0
    └── procfs 0.12.0

error: 3 vulnerabilities found!
```

The last few lines of each summarise the situation:

- Current `main`:
  ```
  error: 9 vulnerabilities found!
  error: 10 denied warnings found!
  ```
- With this PR:
  ```
  error: 3 vulnerabilities found!
  ```

  See #3300 for why we can't resolve all the issues yet.

**Comment by jodh-intel** [comment]
hmmm.... /me wonders if someone unplugged the GH actions server at GH... :smile:

```
Can't find any online and idle self-hosted or hosted runner in the current repository, account/organization that matches the required labels: 'ubuntu-20.04'
Waiting for a self-hosted or a hosted runner to pickup this job...
```

**Comment by Jakob-Naucke** [comment]
@jodh-intel I'm going to abort the current s390x build for maintenance, it wasn't going to be successful anyway. Sorry for the noise.

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks

**Comment by liubin** [comment]
@jodh-intel gnu.org seems to be down :-(

**Comment by jodh-intel** [comment]
Thanks @liubin - that's bad news ;( If it doesn't come back up we may need to force-merge this and/or disable seccomp in the CI's.

**Comment by jodh-intel** [comment]
I've raised https://github.com/kata-containers/kata-containers/pull/3315 to unblock the CI's...

**Comment by jodh-intel** [comment]
Ouch. VFIO build failed as it couldn't find the `protoc` binary:

```
13:30:14 error: failed to run custom build command for `prost-build v0.8.0`
13:30:14 
13:30:14 Caused by:
13:30:14   process didn't exit successfully: `/kata-containers/src/agent/target/release/build/prost-build-cce4214a5591e3b7/build-script-build` (exit status: 101)
13:30:14   --- stderr
13:30:14   thread 'main' panicked at 'Failed to find the protoc binary. The PROTOC environment variable is not set, there is no bundled protoc for this platform, and protoc is not in the PATH', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/prost-build-0.8.0/build.rs:105:10
13:30:14   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
13:30:14 warning: build failed, waiting for other jobs to finish...
13:30:14 error: build failed
13:30:22 make: *** [Makefile:110: target/x86_64-unknown-linux-musl/release/kata-agent] Error 101
```

**Comment by jodh-intel** [comment]
@liubin - any thoughts on why we're seeing this protoc error (now)?

Related: I've updated the agent docs to clarify the API gen commands on #3318.

**Comment by Jakob-Naucke** [comment]
JFYI, `protoc` has been required for building the agent on ppc64le & s390x for as long as I can remember.

**Comment by jodh-intel** [comment]
Aha! After re-reading the CI logs, it now makes sense:

```
16:19:58 error: failed to run custom build command for `prost-build v0.8.0`
```

The problem is that the `prost` agent dependency requires `protoc`. Hence, I'll raise a PR to install `protoc` for all arches, not just ppc64le + s390x.

**Comment by jodh-intel** [comment]
Hmm, Actually, that doesn't explain it fully since the CentOS CI passed for example, but that only install `protobuf-compiler` for ppc64le and s390x...

**Comment by Jakob-Naucke** [comment]
@jodh-intel we install `protobuf-compiler` in many osbuilder containers without paying attention to architecture, it seems VFIO used something else

**Comment by jodh-intel** [comment]
/retest-vfio due to what looks like a network timeout:

```
15:07:27 + /sbin/apk.static -X https://dl-5.alpinelinux.org/alpine/v3.15/main -U --allow-untrusted --root /rootfs --initdb add alpine-base libseccomp
15:07:27 fetch https://dl-5.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
15:07:27 ERROR: https://dl-5.alpinelinux.org/alpine/v3.15/main: temporary error (try again later)
15:07:32 WARNING: Ignoring https://dl-5.alpinelinux.org/alpine/v3.15/main: No such file or directory
15:07:32 ERROR: unable to select packages:
15:07:32   alpine-base (no such package):
15:07:32     required by: world[alpine-base]
15:07:32   libseccomp (no such package):
15:07:32     required by: world[libseccomp]
```

**Comment by fidencio** [comment]
> /retest-vfio due to what looks like a network timeout:
> 
> ```
> 15:07:27 + /sbin/apk.static -X https://dl-5.alpinelinux.org/alpine/v3.15/main -U --allow-untrusted --root /rootfs --initdb add alpine-base libseccomp
> 15:07:27 fetch https://dl-5.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
> 15:07:27 ERROR: https://dl-5.alpinelinux.org/alpine/v3.15/main: temporary error (try again later)
> 15:07:32 WARNING: Ignoring https://dl-5.alpinelinux.org/alpine/v3.15/main: No such file or directory
> 15:07:32 ERROR: unable to select packages:
> 15:07:32   alpine-base (no such package):
> 15:07:32     required by: world[alpine-base]
> 15:07:32   libseccomp (no such package):
> 15:07:32     required by: world[libseccomp]
> ```

You happened to be 2s faster than me. :-)

**Comment by fidencio** [comment]
@jodh-intel, can you rebase your PR atop of `main`?  I would like to rule that out as one of the reasons the rootfs build may be breaking.

**Comment by fidencio** [comment]
This is ready to go, but the kata-deploy build is failing for some reason, and I think a rebase would  solve it.  For now, let me add the `do-not-merge` to avoid it being prematurely merged before James has the chance to rebase the PR.

**Comment by jodh-intel** [comment]
Rebased and pushed. Let's see if we can land this thing in 2021! :smile:

**Comment by fidencio** [comment]
@jongwu, http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/ is red with ARM.

@jodh-intel, whenever you're opening an issue, please, mention http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/

**Comment by ariel-adam** [comment]
Looking at the daily baseline runs for the ARM job in the past 10 days we had 3 failures or 70% success rate:

http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/

@jodh-intel can we try running again the ARM job?

@wainersm what do you think?

**Comment by fidencio** [comment]
> Looking at the daily baseline runs for the ARM job in the past 10 days we had 3 failures or 70% success rate:
> 
> http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/
> 
> @jodh-intel can we try running again the ARM job?

We've been re-running the ARM job since this morning and the last time it passed was Yesterday.
Please, see the build history at http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-ARM-PR/

Screenshot from 2021-12-22 16-01-19

**Comment by fidencio** [comment]
> > Looking at the daily baseline runs for the ARM job in the past 10 days we had 3 failures or 70% success rate:
> > http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/
> > @jodh-intel can we try running again the ARM job?
> 
> We've been re-running the ARM job since this morning and the last time it passed was Yesterday. Please, see the build history at http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-ARM-PR/
> 
> Screenshot from 2021-12-22 16-01-19

And to complete myself here, we even had a green pass of this very same PR in http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-ARM-PR/2963/

Screenshot from 2021-12-22 16-05-12

**Comment by fidencio** [comment]
And now the CI passes. :-).

**Comment by jodh-intel** [comment]
Thanks @fidencio - Christmas came early! ;)

**Comment by ariel-adam** [comment]
Just shows that i should never multi task ;-). 
@wainersm is this still the infra issue causing the ARM job to fail?
I think we said it wasn't an actual issue in their test

**Comment by jongwu** [comment]
> @jongwu, http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/ is red with ARM.
> 
> @jodh-intel, whenever you're opening an issue, please, mention http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-2.0-ubuntu-20.04-ARM-main-baseline/

I think the arm ci fails for the host environmental. I do some cleanup and twist. now I think arm ci will be stable

**Comment by wainersm** [comment]
> Just shows that i should never multi task ;-). @wainersm is this still the infra issue causing the ARM job to fail? I think we said it wasn't an actual issue in their test

@ariel-adam I didn't look at this particular failure of the ARM job; but it is a singular job in the sense that it runs on a bare-metal machine and a previous job execution can interfere with the next if the environment is not properly cleaned up after the tests.

Based on the current historical of executions I'm in favor of setting the ARM job non-required; let it run on a couple of more days and fix problem to then turn it required again. @jongwu - what do you think?

**Pull request merged**